src/lib/CofThy.mli

```diff
@@ -1,57 +1,66 @@
+type cof = (Dim.dim, int) Cof.cof
 type alg_thy
 type disj_thy
-type cof = (Dim.dim, int) Cof.cof
 
-(** Create an empty theory (uses a benign effect for now). *)
-val init : unit -> disj_thy
+module Alg :
+sig
+  type t = alg_thy
 
-(** Returns the consistency of the theory. *)
-val consistency : disj_thy -> [`Consistent | `Inconsistent]
+  (** Create an empty theory (uses a benign effect for now). *)
+  val init : t
 
-(** Assumes the truth of a cofibration; if it can be decomposed eagerly (conjunction of equations),
-    then it does so immediately. Otherwise, it is held "on deck" and repeatedly decomposed when testing
-    sequents. The function [consistency] will consider cofibrations on deck. *)
-val assume : disj_thy -> cof list -> disj_thy
+  (** Returns the consistency of the theory. *)
+  val consistency : t -> [`Consistent | `Inconsistent]
 
-(** Tests the validity of a sequent against the supplied theory. Equivalent to assuming
-    the conjunction of the context and then testing truth. *)
-val test_sequent : disj_thy -> cof list -> cof -> bool
+  (** Construct the enveloping disjunctive theory of an algebraic theory. *)
+  val disj_envelope : alg_thy -> disj_thy
 
-(** Search all branches induced by irreducible joins under additional cofibrations. *)
-val left_invert_under_cofs
-  : zero:'a
-  (** [zero] is the default value for vacuous cases. *)
-  -> seq:((cof -> 'a) -> cof list -> 'a)
-  (** [seq] is the sequencing operator. *)
-  -> disj_thy
-  -> cof list
-  -> (disj_thy -> 'a)
-  -> 'a
+  (** Search all branches induced by irreducible joins under additional cofibrations. *)
+  val left_invert_under_cofs
+    : zero:'a
+    (** [zero] is the default value for vacuous cases. *)
+    -> seq:((t -> 'a) -> t list -> 'a)
+    (** [seq] is the sequencing operator. *)
+    -> t
+    -> cof list
+    -> (t -> 'a)
+    -> 'a
+end
 
-module Algebraic :
+module Disj :
 sig
-  val init : unit -> alg_thy
+  type t = disj_thy
 
-  (** Construct the enveloping disjunctive theory of an algebraic theory. *)
-  val disj_envelope : alg_thy -> disj_thy
+  (** Create an empty theory (uses a benign effect for now). *)
```

Comment:
If this uses a benign effect, how come `init` isn't wrapped in a thunk?

Reply:
LOL that is some leftover from the old comment.